pkg/ccl/backupccl/restore_job.go

```diff
@@ -335,6 +396,9 @@ func restore(
 					if mu.highWaterMark >= 0 {
```

Comment:
could you version gate this? we only need this if we're not doing a 23.1 restore.